osg-comanage-project-usermap.py

```diff
@@ -146,6 +149,35 @@ def get_osguser_groups(filter_group_name=None):
              for pid, gids in pid_gids.items() }
 
 
+def parse_inputfile(inputfile):
+    user_groupmap = dict()
+    with open(inputfile) as file:
+        for line in file:
+            split_line = line.split()
+            if split_line[0] == "*" and len(split_line) >= 3:
+                user_groupmap[split_line[1]] = split_line[2].split(",")
```

Comment:
```suggestion
            # Split up 3 semantic columns
            split_line = line.split(maxsplit=2)
            if split_line[0] == "*" and len(split_line) == 3:
                user_groupmap[split_line[1]] = set(re.split(r'[ ,]+', split_line[2]))
```

I suspect that valid mapping lines allow for combinations of `,` and ` `. For example:

```
* user project1,project2
```

As well as 

```
* user project1, project2
```

We also should use sets so that we don't end up with any dupes in the project list

Reply:
Is there intended behavior for specifying the same user multiple times in the *same* mapfile? Are we alright to overwrite the value in the dict for that user with the last set of projects, or should we also be taking the union of all the projects a user is listed with?

Reply:
I've changed the local inputfile parsing method (now called parse_localmap) to use sets of groups/projects, along with merging all groups for a user if the user is listed multiple times in a projectmap.

Reply:
> Is there intended behavior for specifying the same user multiple times in the same mapfile?

Not sure what HTCondor's behavior would be here but I would expect it to only honor the first line for the user

>  or should we also be taking the union of all the projects a user is listed with?

I would do a union for all sets of projects